clang/lib/DPCT/CallExprRewriterThrust.cpp

```diff
@@ -127,7 +165,7 @@ makeMappedThrustPolicyEnum(unsigned Idx) {
                  getDrefName(CMCE->getArg(0)) + ")";
         }
       }
-    } else if (auto CE = dyn_cast<CallExpr>(E)) {
+    } else if (auto CE = dyn_cast<CallExpr>(C)) {
```

Comment:
why change `E` to `C`?
`C` is `CallExpr`, the `dyn_cast` is unnecessary.
`E` is the argument expression.

Reply:
Good catch, should be E instead. 
while this branch is dead code now, will be removed in the updated PR.